### ISSUE TITLE
queryDevice on HIP is empty

### ISSUE THREAD

**Issue by renderdude** [kokkos/kokkos#4167]
Building on spock with examples on, KokkosExample_query_device returns:
{
}

In the code is:
#if defined(KOKKOS_ENABLE_CUDA)
  Kokkos::Cuda::print_configuration(msg);
#endif

but no HIP equivalent.

**Comment by DavidPoliakoff** [comment]
Should we just replace all of our specialized print_configuration with Kokkos::print_configuration() ? (Question for the team)

**Comment by masterleinad** [comment]
I think 
```
#if defined(KOKKOS_ENABLE_CUDA)
Kokkos::Cuda::print_configuration(msg);
#endif
```
should be replaced by 
```
Kokkos::print_configuration(msg);
```
in all the examples.

**Comment by crtrott** [comment]
Yes and no. Note that ExecSpace::print_configuration is part of our concept for execution spaces. So its always legal to call it. That said the KokkosExample_query_device probably should call the global version.
